### DISCUSSION TITLE
Port forwarding between STA and AP modes.

### DISCUSSION THREAD

**Discussion by Rich968** [espressif/arduino-esp32#10596] [Q&A]
Does the esp32 arduino library support port forwarding/ ip routing between STA and AP modes running simultaneously? Which library should I use for this? ESP8266 had lwip/napt.h for this. Is there anything similar for the esp32?

**Comment by sivar2311** [comment]
What you are looking for is here:  [libraries/WiFi/examples/WiFiExtender/WiFiExtender.ino](https://github.com/espressif/arduino-esp32/blob/master/libraries/WiFi/examples/WiFiExtender/WiFiExtender.ino)

**Reply by Rich968** [reply]
The example does not include port forwarding. eg if a server is connected to the AP it is not possible to reach it from the STA side. To do this the desired STA port and AP port needs to be specified along with ip that the AP has assigned to the server. 

The exam[ple works if you are  connecting to a server on the STA side.
